scanpy/preprocessing/normalization.py

```diff
@@ -0,0 +1,204 @@
+import numpy as np
+from scipy.sparse import issparse
+from sklearn.utils import sparsefuncs
+from .. import logging as logg
+
+def _normalize_data(X, counts, after=None, copy=False):
+    X = X.copy() if copy else X
+    after = np.median(counts) if after is None else after
+    counts += (counts == 0)
+    counts /= after
+    if issparse(X):
+        X = sparsefuncs.inplace_row_scale(X, 1/counts)
+    else:
+        X /= counts[:, None]
+    return X if copy else None
+
+def normalize_quantile(data, counts_per_cell_after=None, counts_per_cell=None,
+                       quantile=1, min_counts=1, key_n_counts=None, copy=False,
+                       layers=[], use_rep=None):
+    """Normalize total counts per cell.
+
+    Normalize each cell by total counts over genes, so that every cell has
+    the same total count after normalization.
+
+    Similar functions are used, for example, by Seurat [Satija15]_, Cell Ranger
+    [Zheng17]_ or SPRING [Weinreb17]_.
+
+    Parameters
+    ----------
+    data : :class:`~anndata.AnnData`
+        The annotated data matrix of shape `n_obs` × `n_vars`. Rows correspond
+        to cells and columns to genes.
+    counts_per_cell_after : `float` or `None`, optional (default: `None`)
+        If `None`, after normalization, each cell has a total count equal
+        to the median of the *counts_per_cell* before normalization.
+    counts_per_cell : `np.array`, optional (default: `None`)
+        Precomputed counts per cell.
+    quantile : `float`, optional (default: 1)
+        Only use genes are less than fraction (specified by *quantile*) of the total
+        reads in every cell.
+    min_counts : `int`, optional (default: 1)
+        Cells with counts less than `min_counts` are filtered out during
+        normalization.
+    key_n_counts : `str`, optional (default: `'n_counts'`)
```

Comment:
Probably, we shouldn't add anything to `adata` by default. There are dedicated functions for doing this in the meanwhile: https://scanpy.readthedocs.io/en/latest/api/scanpy.api.pp.calculate_qc_metrics.html

Also, if allowing for adding something, the naming convention should reflect the discussion around this: https://github.com/theislab/scanpy/pull/358